interpreter/objects/base.py

```
@@ -296,14 +296,18 @@ def call_method(self, name: str, args: list, node=None, code=""):
                 raise
             except TypeError as e:
                 raise QisamJeGhalti(str(e), line, column, code)
+            except IndexError as e:
+                raise IndexJeGhalti(str(e), line, column, code)
```

Ruff (0.16.3)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)